tests/posix/semaphores/prj.conf

```diff
@@ -3,3 +3,7 @@ CONFIG_ZTEST=y
 
 CONFIG_POSIX_AEP_CHOICE_BASE=y
 CONFIG_POSIX_SEMAPHORES=y
+
+CONFIG_DYNAMIC_THREAD=y
+CONFIG_DYNAMIC_THREAD_POOL_SIZE=2
```

Comment:
Same as #88767, we can make it bulletproof by `BUILD_ASSERT`ing  `CONFIG_DYNAMIC_THREAD_POOL_SIZE>=2`

Reply:
Fixed!